core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/rpc/controlreturns.proto

```diff
@@ -0,0 +1,118 @@
+syntax = "proto3";
+package edu.uci.ics.amber.engine.architecture.rpc;
+
+import "edu/uci/ics/amber/engine/architecture/worker/statistics.proto";
+import "scalapb/scalapb.proto";
+
+option (scalapb.options) = {
+  scope: FILE,
+  preserve_unknown_fields: false
+  no_default_values_in_constructor: true
+};
+
+
+// The generic return message
+message ControlReturn {
+  // Oneof block for various return types
+  oneof sealed_value {
+    ControlError error = 1;
+    EmptyReturn emptyReturn = 2;
+    RetrieveWorkflowStateResponse retrieveWorkflowStateResponse = 3;
+    PropagateChannelMarkerResponse propagateChannelMarkerResponse = 4;
+    TakeGlobalCheckpointResponse takeGlobalCheckpointResponse = 5;
+    EvaluatePythonExpressionResponse evaluatePythonExpressionResponse = 6;
+    StartWorkflowResponse startWorkflowResponse = 7;
+    StringResponse stringResponse = 8;
+    IntResponse intResponse = 9;
```

Comment:
Do we have to declare all primitive types as well?

Reply:
I think we should add them whenever they are needed.